src/Networking/MultiplayerManager.cs

```diff
@@ -191,6 +192,19 @@ private static void QueueMainThread(Action action)
             }
         }
 
+        /// <summary>
+        ///     Get the Player object by username. Warning, expensive call!!!
+        /// </summary>
+        public static Player GetPlayerByUsername(string username)
+        {
+            if (username == Instance.CurrentServer.HostPlayer.Username)
+                return Instance.CurrentServer.HostPlayer;
+            else if (Instance.CurrentServer.ConnectedPlayers.Single(z => z.Value.Username == username).Value != null)
```

Comment:
Maybe only execute this call once and just return the Value (since you return null in the else case either way)